dataset.py

```diff
@@ -64,12 +64,12 @@ class Dataset(object):
     When iterated over, this object returns (minibatch, minibatch target) pairs.
     """
     def __init__(self, directory, split=(0.6, 0.2, 0.2), minibatch_size=10,
-            pool_size=8, cache=False):
+            pool_size=8, cache=False, index_file=''):
```

Comment:
`index_file` should be a required argument, since it'll break if it's not supplied.